urlsearchparams/index.html

```diff
@@ -0,0 +1,116 @@
+---
+feature_name: URLSearchParams
+chrome_version: 49
+feature_id: 5632984866619392
+local_css_files: ['page.css']
+additional_head_content: <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Product+Sans">
+---
+
+<h3>Background</h3>
+<p>The <code>URLSearchParams</code> <a href="https://url.spec.whatwg.org/#urlsearchparams" target="_blank">spec</a> defines an interface and convenience methods for working with the query string of a URL (e.g. everything after "<code>?</code>"). This means no more UR regex'ing and string splitting!</p>
+
+<p>A <code>URLSearchParams</code> object can be used directly in a <code>for...of</code> block.</p>
+
+{% capture js %}
+let url = new URL('https://example.com?foo=1&bar=2'); // or construct from window.location
+
+let params = new URLSearchParams(url.search.slice(1));
+
+for (let p of params) {
+  console.log(p);
+}
+
+params.set('baz', 3);
+console.log(params.has('baz'));
+
+params.append('foo', 4);
+console.log(params.getAll('foo'));
+
+console.log(params.toString());
+
+// Update page's URL.
+// window.history.replaceState({}, '', '/?' + params);
+
+{% endcapture %}
+{% include js_snippet.html js=js title="Basic example" %}
+
+{% include output_helper.html %}
+
+<p>The <code>URL</code> constructor integrates with <code>URLSearchParams</code> by adding a read-only property, <code>.searchParams</code>:</p>
+
+{% capture js2 %}
+// Note: .searchParams is currently not implemented in Chrome 49.
+let url = new URL('https://example.com?foo=1&bar=2');
+console.log(url.searchParams.get('foo')); // === 1
+{% endcapture %}
+{% include js_snippet.html js=js2 title="" displayonly=true %}
+
+<h3>Live Demo</h3>
+
+<p>Below, you can manipulate the URL of this page by changing the parameter name
+to get/set values. The URL will update using the <code>URLSearchParams</code> and <code>window.history</code> API.</p>
+
+<div>
+  <label>Parameter: <input type="text" id="paramName" value="foo"></label><label>value: <input type="text" id="paramVal" value="1"></label>
+
+  <div id="radios" class="flex vertical">
+    <label><input type="radio" id="get" name="func" value="get"> get()</label>
+    <label><input type="radio" id="has" name="func" value="has"> has()</label>
+    <label><input type="radio" id="set" name="func" value="set" checked> set()</label>
+    <label><input type="radio" id="getAll" name="func" value="getAll"> getAll()</label>
+    <label><input type="radio" id="append" name="func" value="append"> append()</label>
+    <label><input type="radio" id="delete" name="func" value="delete"> delete()</label>
+  </div>
+
+  <button id="update">Run/update URL</button> <button id="clear">Clear search params</button>
+</div>
+
+<script>
+// Basic sample.
```

Comment:
You don't need all of this if you use `ChromeSamples.log` and `{% capture js %}` instead of `console.log` and this `script`. Have a look at https://github.com/GoogleChrome/samples/blob/gh-pages/report-validity/index.html for instance